docs/architecture/adr-055-orm.md

```diff
@@ -0,0 +1,105 @@
+# ADR 055: ORM
+
+## Changelog
+
+* 2022-04-27: First draft
+
+## Status
+
+ACCEPTED Implemented
+
+## Abstract
+
+In order to make it easier for developers to build state machines and for clients to query, index and verify proofs
+against state data, we have implemented an ORM (object-relational mapping) layer for the Cosmos SDK.
+
+## Context
+
+Historically modules in the Cosmos SDK have always used the key-value store directly and created various handwritten
+functions for managing key format as well as constructing secondary indexes. This consumes a significant amount of
+time when building a module and is error-prone. Because key formats are non-standard, sometimes poorly documented,
+and subject to change, it is hard for clients to generically index, query and verify merkle proofs against state data.
+
+The known first instance of an "ORM" in the Cosmos ecosystem was in [weave](https://github.com/iov-one/weave/tree/master/orm).
+A later version was built for [regen-ledger](https://github.com/regen-network/regen-ledger/tree/master/orm) for
```

Comment:
probably best to use commit hashes instead of `master` within the links of this paragraph, in case this gets removed/outdated